### ISSUE TITLE
{ Enhancement } Elchemy by Example

### ISSUE THREAD

**Issue by ShalokShalom** [wende/elchemy#281]
Hi there :)

I am a newbie, studying the theoretical aspects of programming since about 6 months. 
Elchemy seems like a huuge improvement to other languages in terms of readability and sanity.
My issue is, that the very most tutorials on programming are essentially language documentation.
 
It's like somebody explains to you how to hold a hammer, nails and other tools and says then to you: 

"Now, build a house."

I see empathy as a possible solution. 

Once you can feel and think in the position of the other one, is it far easier to teach. 
Pair programming shows that developers who are much more skilled as their partner, are far less helpful to them. 

I already see the issue appearing up in my own case: 
Things who were unbelievable challenging to me a few months ago are quite fine today. 

I think it's sane to develop tutorials from the point of view from somebody who is actually aware of the current situation for the beginner. 

Besides of this is it for the professional quite annoying, to rid of the little details on the very basic levels over and over again, since it distracts him aways. 

So appears a huge gap, which seems to be practically everywhere ignored. 

In order to make this a story of history, here a proposal: 

I see recipes in cookbooks as a nice approach, since they show how to solve specific tasks. 
Whats even saner, in my opinion, is such a book that takes you by the hand and goes with you through a whole project. Like: How to build a small video game. 

I am really motivated to do something for our community, both Elchemy and the open source scene in general. 

And in order to do so, can I currently offer all kinds of non-programming stuff, so including writing tutorials and creating screencasts of things I am already aware of. 

One of the most helpful guides I experienced is this short series: [The dev aspect](https://www.youtube.com/watch?v=hsTmLhnzRhE&list=PLEoMzSkcN8oNiJ67Hd7oRGgD1d4YBxYGC).

I would love to create such stories and care about this language wherever I can, since honestly: 
I have enough of theory. :)

Elchemy offers me until today the safest option in my opinion since I get an ML-like language which compiles on BEAM. 

To me, it is quite challenging "to start somewhere", since I am theoretically aware about the tools, while I still lack understanding of the different environments, like databases and stuff. 

I am quite aware that there is documentation about them as well, while this suggestion here is about the creation of some tutorials, who have practical value and show the potential of Elchemy in the real world. 

For a newbie.

**Comment by ShalokShalom** [comment]
Leave your time. We can evaluate this together if you want and I am sure more community members are interested in. In general, do I think that static typed and type-inferred languages with a declarative syntax are capable to invite a huge number of newbies into computing. 

Elm is such a language and its capability to run on BEAM is equal to the capability to focus on the happy part of the coding since supervisors can do the job of defensive programming. 

Through that is Elchemy maybe the next Python ;)

**Comment by wende** [comment]
Excuse me the delay.

I think it's a great idea. I definitely agree there is a gap when passing knowledge between people. 
I always liked a theory that a perfect teacher, to explain in the easiest to understand way, is someone with one year more experience than yourself.

I think you might be a really valuable asset to either review or create content that would be understandable for newcomers

If you're up for creating any tutorials you can count on me in terms of providing help, revision and any support required.

Yesterday I've created a Gitbook for Elchemy (https://wende.gitbooks.io/elchemy/content/)
So all of the simple tutorials could go here and feel free to contribute at any time.

If you are thinking about something bigger though than I open for discussion

PS. I think elchemy-empathy is a great code-name 😄

**Comment by ShalokShalom** [comment]
Yeah, I exactly see myself as such one person, who does reviews and contributions to helpful tutorials. 
The question is now, what are our next steps. The huge issue to me is, that I am aware of the tools - and I can always look up for details in the casual documentation - while I feel like clueless, when it comes to writing actual programs. 

What I am really interested about is a documentation about the implementation of a specific piece of software, this is so much more helpful as simple documentation of the language standards. 

Nearly every single piece of the book, tutorial and video shows how to use the languages features as such in theory, while nearly nothing shows the practice. 

"How to code a simple desktop application" "How to code a simple videogame"
"How to code a simple static homepage" "How to ..." You get it.

**Comment by ShalokShalom** [comment]
You can simply record yourself creating such an easy project and explaining why you do this and that, with the assumption that the features of the language are basically understood.

So: How to build an application in practice in Elm.
I would then translate this screencast, so we can upload it within another voice track too and I can also imagine to take screenshots and use the build in subtitle tool in Youtube to automatically transform the spoken content into text, so we can build traditional tutorials as well for those who prefer it instead of videos.
